cheatsheets/Key_Management_Cheat_Sheet.md

```diff
@@ -151,13 +151,15 @@ The generated keys shall be transported (when necessary) using secure channels a
 1. Developers must understand where cryptographic keys are stored within the application. Understand what memory devices the keys are stored on.
 2. Keys must be protected on both volatile and persistent memory, ideally processed within secure cryptographic modules.
 3. Keys should never be stored in plaintext format.
-4. Ensure all keys are stored in cryptographic vault, such as a [hardware security module](https://en.wikipedia.org/wiki/Hardware_security_module) (HSM) or isolated cryptographic service.
+4. Ensure all keys are stored in a cryptographic vault, such as a [hardware security module](https://en.wikipedia.org/wiki/Hardware_security_module) (HSM) or isolated cryptographic service.
 5. If you are planning on storing keys in offline devices/databases, then encrypt the keys using Key Encryption Keys (KEKs) prior to the export of the key material. KEK length (and algorithm) should be equivalent to or greater in strength than the keys being protected.
 6. Ensure that keys have integrity protections applied while in storage (consider dual purpose algorithms that support encryption and Message Code Authentication (MAC)).
 7. Ensure that standard application level code never reads or uses cryptographic keys in any way and use key management libraries.
 8. Ensure that keys and cryptographic operation is done inside the sealed vault.
 9. All work should be done in the vault (such as key access, encryption, decryption, signing, etc).
```

Comment:
This may not always be practical (i.e., scalable) especially with HSM's because of costs. Cloud-based KMS's are probably a good alternative for those storing their encrypted data in the cloud.